Revit_Core_Engine/Convert/Revit/FromRevit/View.cs

```diff
@@ -0,0 +1,164 @@
+            if (view != null)
+                return view;
+
+            if (revitView.AreGraphicsOverridesAllowed()) { 
+                /*1. Transfer ViewFilters and corresponding OverrideGraphicSettings into ViewFilterWithOverrides Objects - via STREAMS */
+                List<ViewFilterWithOverrides> filtersWithOverrides = revitView.GetFilters().ToDictionary<ElementId, ViewFilter, OverrideGraphicSettings>
+                                                            (elId => Convert.ViewFilterFromRevit((ParameterFilterElement)revitView.Document.GetElement(elId)),
+                                                                elId => overrideGraphicSettingsFromRevit(revitView, revitView.GetFilterOverrides(elId))).ToList()
+                                                                .Select(kvp => new ViewFilterWithOverrides { Filter = kvp.Key, Overrides = kvp.Value }).ToList();
+                /*2. Create BHoM View Object with Name and FilterWithOverrides objects */
+                view = new oM.Adapters.Revit.Elements.View { Name = revitView.Name, FiltersWithOverrides = filtersWithOverrides };}
+            else { view = new oM.Adapters.Revit.Elements.View { Name=revitView.Name };}
+
+            //Set identifiers, parameters & custom data
+            view.SetIdentifiers(revitView);
+            view.CopyParameters(revitView, settings.MappingSettings);
+            view.SetProperties(revitView, settings.MappingSettings);
+
+            refObjects.AddOrReplace(revitView.Id, view);
+            return view;
+        }
+
+        /***************************************************/
+
+        private static OverrideGraphicSettings overrideGraphicSettingsFromRevit(this View element, Autodesk.Revit.DB.OverrideGraphicSettings revitOverrides)
+        {
+            // Initialize BHoM OverrideGraphicsSettings object
+            OverrideGraphicSettings overrideGraphicsSettings = new OverrideGraphicSettings();
+
+
+            // Convert COLORS 
+            if (revitOverrides.CutLineColor.IsValid) overrideGraphicsSettings.LineColor = System.Drawing.Color.FromArgb(revitOverrides.CutLineColor.Red, revitOverrides.CutLineColor.Green, revitOverrides.CutLineColor.Blue);
+            if (revitOverrides.CutForegroundPatternColor.IsValid) overrideGraphicsSettings.CutColor = System.Drawing.Color.FromArgb(revitOverrides.CutBackgroundPatternColor.Red, revitOverrides.CutBackgroundPatternColor.Green, revitOverrides.CutBackgroundPatternColor.Blue);
+            if (revitOverrides.SurfaceBackgroundPatternColor.IsValid) overrideGraphicsSettings.SurfaceColor = System.Drawing.Color.FromArgb(revitOverrides.SurfaceBackgroundPatternColor.Red, revitOverrides.SurfaceBackgroundPatternColor.Green, revitOverrides.SurfaceBackgroundPatternColor.Blue);
+
+            // Convert LINE PATTERNS
+            try 
+                { 
+                    String linePatternName = element.Document.GetElement(revitOverrides.CutLinePatternId).Name;
```

Comment:
Instead of wrapping all of that in try/catch it should be enough to check if `revitOverrides.CutLinePatternId != -1`. Or am I missing a point?

Reply:
@pawel, thanks for this. 
No specific reason to go for try/catch rather than if/then statement.
Updated the code in line with your comments.

Reply:
FYI landing in `catch` is extremely slow - it triggers the runtime to do reporting, generate stack trace, dump memory etc. For that reason it should be last resort rather than an alternative to ifs 👍